doc/release/README.executing.rst

```diff
@@ -0,0 +1,369 @@
+execute a program is specified on the executable's command-line.  This
+can be set either using the -nl flag, or by assigning to the built-in
+numLocales configuration constant using the normal mechanisms.  So, to
+execute on four locales, one could use:
+
+  .. code-block:: sh
+
+    ./a.out -nl 4
+
+or:
+
+  .. code-block:: sh
+
+    ./a.out --numLocales=4
+
+or:
+
+  .. code-block:: sh
+
+    ./a.out -snumLocales=4
+
+For users running with ``$CHPL_COMM=none`` (the default), only one
+locale can be used.  See :doc:`multilocale` for more
+information about executing on multiple locales.
+
+Multi-locale programs often use a launcher executable to do some initial
+command-line checking before spawning the real program, which is then
+stored in a second binary named *original_binary_name*\ ``_real``.  See
+:doc:`launcher` for more information about the launcher executable.
+
+
+--------------------------------------
+Controlling Degree of Data Parallelism
+--------------------------------------
+
+Data parallel operations over ranges, default domains, and default
+arrays permit the number of tasks used to implement the data
+parallelism to be specified using the following built-in configuration
+constants:
+
+  ``dataParTasksPerLocale``
+    Number of Chapel tasks to use to execute forall loops (default:
+    number of physical CPUs on the node, which may be reduced by
+    ``CHPL_RT_NUM_THREADS_PER_LOCALE``.  When ``CHPL_TASKS=qthreads``,
+    ``CHPL_RT_NUM_THREADS_PER_LOCALE`` can also increase this, up to the
+    number of logical CPUs).
+
+  ``dataParIgnoreRunningTasks``
+    If ``true``, always use ``dataParTasksPerLocale`` tasks to execute
+    forall loops.  If ``false``, reduce the number of tasks used by the
+    number of tasks already running (default: ``false``).
+
+  ``dataParMinGranularity``
+    The number of tasks used to execute forall loops should be reduced
+    such that the number of iterations per task is never less than the
+    specified value (default: ``1``).
+
+Most Chapel standard distributions also use identically named
+constructor arguments to control the degree of data parallelism within
+each locale when iterating over its domains and arrays.  The default
+values for these arguments are set to the corresponding global
+configuration constants.
+
+
+-------------------------------------------------
+Environment Variable Control over Chapel Behavior
+-------------------------------------------------
+
+Chapel uses environment variables to control the number of threads used
+at execution time and the call stack size, among other things.  In many
+cases third-party packages used by Chapel define their own environment
+variables to provide the same or similar control.  For Chapel programs,
+Chapel environment variables always take precedence over third-party
+ones when both control the same thing.  However, a third-party setting
+will still override a Chapel default, if those differ.
```

Comment:
The shortest version is:
* if the Chapel env var is set then that is the value
* otherwise if the 3rd-party package env var is set then that is the value
* otherwise the Chapel default is the value.